hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/record/SingleTransactionRecordBuilder.java

```diff
@@ -18,17 +18,62 @@
 
 import static com.hedera.node.app.spi.workflows.HandleContext.TransactionCategory.USER;
 
+import com.hedera.hapi.node.base.AccountAmount;
+import com.hedera.hapi.node.base.AccountID;
 import com.hedera.hapi.node.base.ResponseCodeEnum;
 import com.hedera.hapi.node.base.Transaction;
+import com.hedera.hapi.node.base.TransactionID;
+import com.hedera.hapi.node.transaction.ExchangeRateSet;
 import com.hedera.hapi.node.transaction.SignedTransaction;
 import com.hedera.hapi.node.transaction.TransactionBody;
 import com.hedera.node.app.spi.workflows.HandleContext;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
 import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Instant;
+import java.util.List;
+import java.util.Set;
 
 /**
- * This interface contains methods to read general properties a transaction record builder.
+ * Defines API for constructing stream items of a single transaction dispatch.
+ * The implementation may produce only records or could produce block items
  */
 public interface SingleTransactionRecordBuilder {
+    /**
+     * Sets the transaction for this stream item builder.
+     * @param transaction the transaction
+     * @return this builder
+     */
+    SingleTransactionRecordBuilder transaction(@NonNull Transaction transaction);
+
+    /**
+     * Returns the transaction for this stream item builder.
+     * @return the transaction
+     */
+    Transaction transaction();
+
+    /**
+     * Returns any ids recorded as having explicit reward situations during the construction of this builder so far.
+     * @return the ids with explicit reward situations
+     */
+    Set<AccountID> explicitRewardSituationIds();
```

Comment:
It is related to Staking rewards in RecordFinalizer, because this is how contract service signals that called contracts have special reward situations for mono-service compatibility. This could be changed in the future if we decide to stop simulating mono-service